.github/workflows/lint-build-commits.yaml

```diff
@@ -107,7 +163,13 @@ jobs:
           COMMITS=$(git rev-list ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }})
           for commit in $COMMITS ; do
             git checkout $commit || exit 1
-            (make -C test build -j $(nproc) && make -C test build-darwin -j $(nproc)) || exit 1
+            # Do not run make if there aren't any files modified in these
+            # directories from the previous commit to the current commit.
+            # If these filters are modified, also modify the step above where we
+            # run dorny/paths-filter.
+            if ! git diff --quiet HEAD^ pkg/ test/ ; then
```

Comment:
The check for test here is somewhat redundant, since the step already only runs if there are changes in test (at least, that's what the comment claims). That, though, also seems broken - a change in an API (say, in `pkg/foo`) used in the ginkgo test would not be noticed?

Reply:
I'm not following. Same answer as I gave for the other comment but in this case would be the if statement on line 160

Reply:
This one definitely has higher risk than the BPF one, though I would also argue that ideally the gingko tests should generally avoid linking directly against internal implementation details from the agent libraries. That said, I think that there are relatively few dependencies from `test/` into `pkg/` currently so hopefully this isn't too big of an issue.